lib/chef/mixin/powershell_exec.rb

```diff
@@ -90,22 +96,35 @@
 class Chef
   module Mixin
     module PowershellExec
-      # Run a command under PowerShell via a managed (.NET) COM interop API.
-      # This implementation requires the managed dll to be registered on the
-      # target machine.
+      # Run a command under PowerShell via a managed (.NET) API.
       #
       # Requires: .NET Framework 4.0 or higher on the target machine.
       #
       # @param script [String] script to run
+      # @param interpreter [Symbol] the interpreter type, `:powershell` or `:pwsh`
       # @return [Chef::PowerShell] output
-      def powershell_exec(script)
-        Chef::PowerShell.new(script)
+      def powershell_exec(script, interpreter = :powershell)
+        case interpreter
+        when :powershell
+          Chef::PowerShell.new(script)
+        when :pwsh
+          Chef::Pwsh.new(script)
+        else
+          raise ArgumentError, "Expected interpreter of :powershell or :pwsh"
+        end
       end
 
       # The same as the #powershell_exec method except this will raise
       # Chef::PowerShell::CommandFailed if the command fails
-      def powershell_exec!(script)
-        cmd = Chef::PowerShell.new(script)
+      def powershell_exec!(script, interpreter = :powershell)
+        case interpreter
+        when :powershell
+          cmd = Chef::PowerShell.new(script)
+        when :pwsh
+          cmd = Chef::Pwsh.new(script)
+        else
+          raise ArgumentError, "Expected interpreter of :powershell or :pwsh"
+        end
```

Comment:
```suggestion
        cmd = powershell_exec(script, interpreter)
```

Any reason to not lean on the other method?

Reply:
oh of course!